habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/util/TomlUtils.java

```diff
@@ -79,7 +79,20 @@ public final class TomlUtils {
      * e.g. "[a.b]" or "[[a.b]]", capturing the text between the brackets
      * as group 1 (the full section name without surrounding '[' or ']').
      */
-    private static final String TOML_SECTION_HEADER_REGEX = "^\\[+(.+?)\\]+$";
+    private static final Pattern TOML_SECTION_HEADER_PATTERN = Pattern.compile("^\\[+(.+?)\\]+$");
+
+    /**
+     * Matches an inline Python dependency assignment in a TOML table
+     * using single/double quotes, and with optional whitespaces
+     * e.g. `python = ">=3.12.0,<4"`, `Python='^3.12'`
+     */
+    public static final Pattern PYTHON_DEPENDENCIES_PATTERN = Pattern.compile("(?i)\\bpython\\s*=\\s*(?:'[^']*'|\"[^\"]*\")");
```

Comment:
A: couldn't you just do
```suggestion
    public static final Pattern PYTHON_DEPENDENCIES_PATTERN = Pattern.compile("\\bpython\\s*=\\s*('|\").*?('|\")");
```
Why do we need case-insensitive?

Reply:
yes, you're right! no need for case-insensitive & simplified the regex as you suggested.